code/modules/transport/pods/MainWeapon.dm

```diff
@@ -380,6 +380,608 @@
+			mode = EFIF_MODE_REPAIR
+			firerate = 15
+
+		else if(href_list["cutter"])
+			mode = EFIF_MODE_DRILL
+			firerate = 12
+
+		else if(href_list["wide_field"])
+			wide_field = !wide_field
+
+		else if(href_list["load"])
+			src.scoop_up_sheets()
+
+		opencomputer(usr)
+		return
+
+/obj/overlay/construction_field
+	name = "energy"
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "buildeffect"
+	layer = EFFECTS_LAYER_BASE
+	///Tracks the build type of what each field is building for cost calculation and placement logic
+	var/to_build
+	///If in repair mode, build the precise initial turf
+	var/repair_mode = FALSE
+
+/datum/action/bar/construction_field
+	var/obj/machinery/vehicle/pod
+	var/obj/item/shipcomponent/mainweapon/constructor/buildtool
+	var/mode
+	///Sum of current attempted construction's cost in sheets
+	var/action_build_cost = 0
+	duration = 1 SECONDS
+
+	New(var/obj/thepod,var/obj/thetool,var/passmode)
+		src.pod = thepod
+		src.buildtool = thetool
+		src.mode = passmode
+		for (var/obj/overlay/construction_field/F in buildtool.active_fields)
+			src.duration += src.get_duration_mod(F)
+			src.action_build_cost += buildtool.get_cost_mod(F)
+		..()
+
+	proc/get_duration_mod(var/obj/overlay/construction_field/F)
+		switch(F.to_build)
+			if(EFIF_MODE_FLOORS)
+				return EFIF_FLOOR_BUILD_TIME
+			if(EFIF_MODE_R_FLOORS)
+				return EFIF_R_FLOOR_BUILD_TIME
+			if(EFIF_MODE_WALLS)
+				var/space_adjustment = EFIF_WALL_BUILD_TIME
+				if(istype(get_turf(F),/turf/space))
+					space_adjustment += EFIF_FLOOR_BUILD_TIME
+				return space_adjustment
+
+	///Check whether dense objects are present in build fields; interrupt any fields which have been blocked
+	proc/dense_object_refresh()
+		. = FALSE
+		for (var/obj/overlay/construction_field/field in buildtool.active_fields)
+			for (var/obj/O in get_turf(field))
+				if (O.density && !(istype(O, /obj/structure/girder)))
+					src.action_build_cost -= buildtool.get_cost_mod(field)
+					buildtool.active_fields -= field
+					qdel(field)
+					break
+
+	onStart()
+		..()
+		//didn't work right for some reason
+		/*
+		if(pod.bound_height == 64)
+			bar.pixel_x += 16
+			bar.pixel_y += 16
+			border.pixel_x += 16
+			border.pixel_y += 16
+			updateBar()
+		*/
+
+		if (buildtool.steel_sheets < action_build_cost)
+			interrupt(INTERRUPT_ALWAYS)
+			return
+
+		src.dense_object_refresh()
+
+		if (!length(buildtool.active_fields))
+			interrupt(INTERRUPT_ALWAYS)
+			return
+
+	onInterrupt(var/flag)
+		for(var/atom/field in buildtool.active_fields)
+			qdel(field)
+		buildtool.active_fields = list()
+		..()
+
+	onDelete()
+		for(var/atom/field in buildtool.active_fields)
+			qdel(field)
+		buildtool.active_fields = list()
+		..()
+
+	onEnd()
+		..()
+		src.dense_object_refresh()
+		if (buildtool.steel_sheets < action_build_cost || !length(buildtool.active_fields))
+			interrupt(INTERRUPT_ALWAYS)
+			return
+		///Plural management for log formatting
+		var/multy = FALSE
```

Comment:
Typo?

Reply:
Mult-y - has the quality of multipleness. I liked it because it's a little silly, but if something more formal is preferred I'm not that attached to it.